Core/include/Acts/Utilities/ScopedTimer.hpp

```diff
@@ -0,0 +1,70 @@
+// This file is part of the ACTS project.
+//
+// Copyright (C) 2016 CERN for the benefit of the ACTS project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Utilities/Logger.hpp"
+
+#include <chrono>
+#include <iostream>
+
+namespace Acts {
+
+/// @brief A RAII timer class for measuring execution time of code blocks
+///
+/// ScopedTimer provides automatic timing of code blocks using RAII principles.
+/// It starts timing when constructed and automatically logs the duration when
+/// destroyed. This makes it ideal for measuring execution time of functions
+/// or code blocks without manual start/stop calls.
+///
+/// Example usage:
+/// @code
+/// {
+///   ScopedTimer timer("myFunction");
+///   // ... code to measure ...
+/// } // Timer automatically logs duration when block ends
+/// @endcode
+///
+class ScopedTimer {
+ public:
+  /// @brief Construct a new Scoped Timer
+  ///
+  /// @param name Identifier for the timed block
+  /// @param lvl Logging level for the timing output
+  /// @param logger Logger instance to use for output
+  explicit ScopedTimer(const std::string& name,
+                       Logging::Level lvl = Logging::Level::INFO,
```

Comment:
It could in some cases be more convenient to have the arguments ordered the other way around, because to actual get any output, you'll have to override the third argument, and by doing so always the second...
But this is very nitpicky of course.